crates/holochain/src/conductor/handle.rs

```diff
@@ -389,16 +409,20 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
                 respond.respond(Ok(async move { Ok(signature) }.boxed().into()));
             }
             _ => {
-                let cell = self.cell_by_id(cell_id).await?;
-                cell.handle_holochain_p2p_event(event).await?;
+                self.with_cell(cell_id, |cell| {
+                    async move { Ok(cell.handle_holochain_p2p_event(event).await?) }.boxed()
+                })
+                .await?
```

Comment:
Every network message will now lock the conductor. This is what we changed before because it causes dead locks when something requires a write lock while a network message or zome call is in progress.